### PR TITLE
chore(config): Update field labels for commonly used sources and transforms 

### PR DESCRIPTION
**Pull request by maycmlee**
This PR updates the commonly used sources and transforms' field labels that needs a human_name label. Similar to this [PR](https://github.com/vectordotdev/vector/pull/17475).

### CODE REVIEW

File: src/transforms/remap.rs [hunk 1]
```diff
@@ -129,6 +130,7 @@ pub struct RemapConfig {
     /// to a specially-named output, `dropped`. The original event is annotated with additional
     /// fields describing why the event was dropped.
     #[serde(default = "crate::serde::default_false")]
+    #[configurable(metadata(docs::human_name = "Drop on Abort"))]
```
Comment (tobz) [inline src/transforms/remap.rs:133]: Looks like this should be applied to `drop_on_abort` instead?

File: src/transforms/aggregate.rs [hunk 2]
```diff
@@ -26,6 +26,7 @@ pub struct AggregateConfig {
     ///
     /// During this time frame, metrics with the same series data (name, namespace, tags, and so on) are aggregated.
     #[serde(default = "default_interval_ms")]
+    #[configurable(metadata(docs::human_name = "Interval Time Frame"))]
```
Comment (tobz) [inline src/transforms/aggregate.rs:29]: "Interval Time Frame" feels a bit repetitive, but it also doesn't fit the pattern of "X Interval" that similar fields have been changed to.

How about "Flush Interval"?

File: src/sources/util/multiline_config.rs [hunk 3]
```diff
@@ -18,6 +18,7 @@ pub struct MultilineConfig {
     #[configurable(metadata(docs::examples = "\\\\$"))]
     #[configurable(metadata(docs::examples = "^(INFO|ERROR) "))]
     #[configurable(metadata(docs::examples = ";$"))]
+    #[configurable(metadata(docs::human_name = "Timeout"))]
```
Comment (tobz) [inline src/sources/util/multiline_config.rs:21]: Mistakenly copypasted entirely, or just never updated for the field at hand?

File: src/sources/splunk_hec/acknowledgements.rs [hunk 4]
```diff
@@ -27,28 +27,32 @@ pub struct HecAcknowledgementsConfig {
...
     /// Equivalent to the `max_number_of_acked_requests_pending_query` Splunk HEC setting.
     ///
     /// Minimum of `1`.
+    #[configurable(metadata(docs::human_name = "Max Number of Pending Acknowledgements"))]
     pub max_pending_acks: NonZeroU64,
 
     /// The maximum number of Splunk HEC channels clients can use with this source.
     ///
     /// Minimum of `1`.
+    #[configurable(metadata(docs::human_name = "Max Number of  Acknowledgement Channels"))]
```
Comment (tobz) [inline src/sources/splunk_hec/acknowledgements.rs:41]: Double space between `of` and `Acknowledgement`.

File: src/sources/kafka.rs [hunk 5]
```diff
@@ -107,26 +107,31 @@ pub struct KafkaSourceConfig {
     #[configurable(metadata(docs::examples = 5000, docs::examples = 10000))]
     #[configurable(metadata(docs::advanced))]
     #[serde(default = "default_session_timeout_ms")]
+    #[configurable(metadata(docs::human_name = "Session Timeout"))]
+
```
Comment (tobz) [inline src/sources/kafka.rs:111]: Drop this extra empty line.

**Review by tobz — CHANGES_REQUESTED** [review]

File: src/sources/util/multiline_config.rs (hunk 3 above)
Comment (maycmlee) [reply src/sources/util/multiline_config.rs:21]: Oops, that was a mistake.

File: src/transforms/aggregate.rs (hunk 2 above)
Comment (maycmlee) [reply src/transforms/aggregate.rs:29]: "Flush Interval" is much better!

**Comment by maycmlee** [comment]
Thanks @tobz for catching those issues! 🙃 They should all be fixed.

File: src/sources/opentelemetry/mod.rs [hunk 6]
```diff
@@ -68,6 +68,7 @@ pub struct OpentelemetryConfig {
 #[configurable(metadata(docs::examples = "example_grpc_config()"))]
 #[derive(Clone, Debug)]
 #[serde(deny_unknown_fields)]
+#[configurable(metadata(docs::human_name = "gRPC"))]
```
Comment (silversupreme) [inline src/sources/opentelemetry/mod.rs:71]: Should this be added to `WELL_KNOWN_REPLACEMENTS`?

File: src/transforms/remap.rs [hunk 7]
```diff
@@ -129,6 +130,7 @@ pub struct RemapConfig {
     /// to a specially-named output, `dropped`. The original event is annotated with additional
     /// fields describing why the event was dropped.
     #[serde(default = "crate::serde::default_false")]
+    #[configurable(metadata(docs::human_name = "Drop on Error"))]
```
Comment (silversupreme) [inline src/transforms/remap.rs:133]: Should this be `Reroute Dropped Output` or similar?

**Review by silversupreme — APPROVED** [review]
6j67p7

File: src/sources/opentelemetry/mod.rs (hunk 6 above)
Comment (maycmlee) [reply src/sources/opentelemetry/mod.rs:71]: Oh yes, and actually, it's already there. I'll remove this addition.

File: src/sources/kafka.rs [hunk 8]
```diff
@@ -107,26 +107,30 @@ pub struct KafkaSourceConfig {
...
     #[serde(default = "default_socket_timeout_ms")]
+    #[configurable(metadata(docs::human_name = "Socket Timeout"))]
     socket_timeout_ms: Duration,
 
     /// Maximum time the broker may wait to fill the response.
     #[serde_as(as = "serde_with::DurationMilliSeconds<u64>")]
     #[configurable(metadata(docs::examples = 50, docs::examples = 100))]
     #[configurable(metadata(docs::advanced))]
     #[serde(default = "default_fetch_wait_max_ms")]
+    #[configurable(metadata(docs::human_name = "Fetch Wait Max"))]
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply src/sources/kafka.rs]: Good call, updated.

File: src/sources/splunk_hec/acknowledgements.rs (hunk 4 above)
Comment (maycmlee) [reply src/sources/splunk_hec/acknowledgements.rs]: Fixed

File: src/sources/kafka.rs (hunk 5 above)
Comment (maycmlee) [reply src/sources/kafka.rs]: Fixed

File: src/transforms/remap.rs (hunk 1 above)
Comment (maycmlee) [reply src/transforms/remap.rs]: Fixed.

File: src/transforms/remap.rs [hunk 9]
```diff
@@ -117,6 +117,7 @@ pub struct RemapConfig {
     ///
     /// [vrl_docs_abort]: https://vector.dev/docs/reference/vrl/expressions/#abort
     #[serde(default = "crate::serde::default_true")]
+    #[configurable(metadata(docs::human_name = "Drop on Error"))]
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply src/transforms/remap.rs]: Fixed

File: src/sources/aws_kinesis_firehose/mod.rs [hunk 10]
```diff
@@ -37,15 +37,15 @@ pub struct AwsKinesisFirehoseConfig {
     #[configurable(metadata(docs::examples = "localhost:443"))]
     address: SocketAddr,
 
-    /// An optional access key to authenticate requests against.
+    /// An access key to authenticate requests against.
     ///
     /// AWS Kinesis Firehose can be configured to pass along a user-configurable access key with each request. If
     /// configured, `access_key` should be set to the same value. Otherwise, all requests are allowed.
     #[configurable(deprecated = "This option has been deprecated, use `access_keys` instead.")]
     #[configurable(metadata(docs::examples = "A94A8FE5CCB19BA61C4C08"))]
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply src/sources/aws_kinesis_firehose/mod.rs:45]: That would also drop it from the Vector docs - which might not be desired until we actually remove it. That said, it's probably been deprecated for more than a release and can be removed now.

File: src/transforms/remap.rs (hunk 7 above)
Comment (spencergilbert) [reply src/transforms/remap.rs:133]: Agreed, I think this line should be up at https://github.com/vectordotdev/vector/pull/17517/files#diff-ad5088e389a2b5ae3b514738658923d21b23dda2de8b29cfb4022c73cd0f8bd2L105

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: src/transforms/remap.rs (hunk 7 above)
Comment (maycmlee) [reply src/transforms/remap.rs]: @silversupreme @spencergilbert I think I finally got this right. I used `Reroute Dropped Events`. Added `Drop Event on Error` to up where it should be. And also updated the `Drop on Abort` to `Drop Event on Abort`. Let me know if that looks good, thanks!

File: src/sources/aws_kinesis_firehose/mod.rs (hunk 10 above)
[the remark this answers is not in the document]
Comment (maycmlee) [reply src/sources/aws_kinesis_firehose/mod.rs:45]: I have a list of fields that are noted as deprecated. This is one of them, so I'm thinking we can address them all in one PR?

**Review by spencergilbert — APPROVED** [review]

**Review by tobz — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**